### PR TITLE
Send log-event if worker is restarted because of memory pressure

### PR DESCRIPTION
**Pull request by phofl**
Closes #xxxx

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/worker_memory.py [hunk 1]
```diff
@@ -417,12 +417,14 @@ def memory_monitor(self, nanny: Nanny) -> None:
...
-            nanny_logger.warning(
+            msg = (
                 f"Worker {nanny.worker_address} (pid={process.pid}) exceeded "
                 f"{self.memory_terminate_fraction * 100:.0f}% memory budget. "
-                "Restarting...",
+                f"Restarting..."
             )
+            nanny_logger.warning(msg)
             self._last_terminated_pid = process.pid
+            nanny.log_event("worker-restart-memory", msg)
```
Comment (fjetter) [inline distributed/worker_memory.py:427]: The full text message is only relevant for the logger but the `log_event` stuff is supposed to be machine readable and prose will not help us much.

```suggestion
            msg = {
                "worker": nanny.worker_address,
                 "pid": process.pid,
                 "rss": memory,
            }
            nanny.log_event("worker-restart-memory", msg)
```

This is easier to parse automatically and contains the same / more information that the long string.

Comment (phofl) [reply distributed/worker_memory.py:427]: Thx for the explanation, sounds good to me

File: distributed/tests/test_nanny.py [hunk 2]
```diff
@@ -660,6 +660,10 @@ async def test_restart_memory(c, s, n):
     while not s.workers:
         await asyncio.sleep(0.1)
 
+    assert all(
+        "memory budget" in msg[1] for msg in s.get_events("worker-restart-memory")
+    )
+
```
Comment (fjetter) [inline distributed/tests/test_nanny.py:666]:
```suggestion
    msgs = s.get_events("worker-restart-memory")
    assert len(msgs) == 1
    msg = msgs[0][1]
    assert isinstance(msg, dict)
    assert {"worker", "pid", "rss"}.issubset(set(msg))

```

**Comment by phofl** [comment]
thx, ci seemed fine

**Pull request merged**